### DISCUSSION TITLE
CommandFInished exitCode 0 when generators outputs "already exist"

### DISCUSSION THREAD

**Discussion by dillingham** [laravel/framework#36072] [Q&A]
Not sure if this is intended or a bug.. but CommandFinished event for console outputs exitCode 0 (success) when the commands for `make` (or atleast make:policy) when it wasnt successful. How do we early return from a listener if the command didnt execute as intended? I even tried `Artisan::output()` from the listener but it doesnt output "Policy already exists!" which would be a workaround at best

**Comment by ahinkle** [comment]
<img width="214" alt="Screen Shot 2021-02-02 at 8 46 58 PM" src="https://user-images.githubusercontent.com/17038330/106690699-e61c1800-6597-11eb-9edf-0faa995dfaa3.png">
Hmm..

**Reply by dillingham** [reply]
Yeah it correctly prevents a second policy, but the event `CommandFinished` from Laravel has a $exitCode = 0 when `Policy already exists!` Which I guess indicates the command was "successful" when it wasnt
